doc/source/contributing_docs.rst

```diff
@@ -103,6 +104,63 @@ Saving state in Jupyter notebooks
 
 One of the neat things about ``nglview`` is the ability to interact with molecules via the viewer. This ability can be preserved for the HTML pages generated from Jupyer notebooks by ``nbsphinx``, if you save the notebook widget state after execution.
 
+.. _nbval-testing:
+
+Test with pytest and nbval
+===========================
+
+Whenever you add or modify notebook cells, you should make sure they run without errors and that their
+outputs are consistent, since they are part of the documentation as well.
+
+We use a pytest plugin for this called `nbval`_, it takes advantage of the saved notebook state
+and re-runs the notebook to determine if its output is still identical to the saved state.
+Thus, cells not only have to work (no errors), but also must give the same output they gave when
+they were saved.
+
+To test all notebooks, just cd into ``UserGuide/tests`` and run ``pytest``.
+If you want to test a particular notebook, check the the contents of `pytest.ini`, this file
+defines flags that you can also pass directly to pytest.
+For example, if you wanted to test the `hole2` notebook::
+
+    pytest --nbval --nbval-current-env --nbval-sanitize-with ./sanitize_output.cfg ../doc/source/examples/analysis/polymers_and_membranes/hole2.ipynb
+Where ``--nbval`` tells pytest to use nbval to test Jupyter notebooks, ``--nbval-current-env``
+to use the currently loaded python environment (make sure you actually loaded your environment)
+instead of trying to use the one that was used when the notebook was saved  and
+``--nbval-sanitize-with`` to point pytest to a file full of replacement rules like this one
+for example::
+
+    regex: (.*B \[0.*B/s\])
+    replace: DOWNLOAD
+This tells pytest to scan the outputs of all cells and replace the matching string with the word
+*DOWNLOAD*. This is called *sanitization*.
+
+.. _`nbval`: https://nbval.readthedocs.io/en/latest/
+
+sanitization
+""""""""""""
+Exactly matching cell outputs between runs is a high bar for testing and tends to give false positives.
+Otherwise correct cells may give different outputs each time they are run (e.g. cells with code
```

Comment:
```suggestion
Exactly matching cell outputs between runs is a high bar for testing and tends to give false errors
-- otherwise correct cells may give different outputs each time they are run (e.g. cells with code
```

I understand what you mean here, but to me a "positive" result when testing would be no tests!